### PR TITLE
Configuring targets for running uses internal methods allowing commad line tools and apps to be launchable.

### PR DESCRIPTION
**Pull request by dostrander**
@segiddins @dnkoutso

### CODE REVIEW

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* Configuring targets for running uses internal methods allowing commad line tools and apps to be executable.  
  [dostrander](https://github.com/dostrander)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by dnkoutso** [comment]
lgtm, is it possible to add tests?

**Comment by dostrander** [comment]
YEah i'll add some tests

File: CHANGELOG.md [hunk 1]
```diff
@@ -65,7 +68,9 @@
 
 ##### Enhancements
 
-* None.  
+* Add support for launchable targets from a scheme.
```
Comment (segiddins) [inline CHANGELOG.md]: duplicated entry

File: lib/xcodeproj/project/object/native_target.rb [hunk 2]
```diff
@@ -460,6 +460,15 @@ def extension_target_type?
           end
         end
 
+        def launchable_target_type?
```
Comment (segiddins) [inline lib/xcodeproj/project/object/native_target.rb]: docs for this method

File: lib/xcodeproj/scheme.rb [hunk 3]
```diff
@@ -64,13 +64,15 @@ def initialize(file_path = nil)
...
     #        The target to use for the 'Test' action
     #
-    def configure_with_targets(runnable_target, test_target)
-      build_action.add_entry BuildAction::Entry.new(runnable_target) if runnable_target
-      build_action.add_entry BuildAction::Entry.new(test_target) if test_target
-
-      test_action.add_testable TestAction::TestableReference.new(test_target) if test_target
-      launch_action.buildable_product_runnable = BuildableProductRunnable.new(runnable_target, 0) if runnable_target
-      profile_action.buildable_product_runnable = BuildableProductRunnable.new(runnable_target) if runnable_target
+    def configure_with_targets(runnable_target, test_target, launch_target: false)
```
Comment (segiddins) [inline lib/xcodeproj/scheme.rb]: docs for the `launch_target` parameter

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,9 @@
 
 ##### Enhancements
 
+* Add support for launchable targets from a scheme.  
+  [Derek Ostrander](https://github.com/dostrander)
+  [#614](https://github.com/CocoaPods/Xcodeproj/pull/614)
```
Comment (dnkoutso) [inline CHANGELOG.md]: nit: press enter

File: lib/xcodeproj/scheme.rb [hunk 5]
```diff
@@ -64,13 +64,18 @@ def initialize(file_path = nil)
...
+    # @param [Boolean] launch_target
+    #        Determines if the runnable_target is launchable.
+    #
+    def configure_with_targets(runnable_target, test_target, launch_target: false)
+      if runnable_target
+        add_build_target(runnable_target)
+        set_launch_target(runnable_target) if launch_target
+      end
+      if test_target
+        add_build_target(test_target, false) if test_target != runnable_target
```
Comment (joshuapoq) [inline lib/xcodeproj/scheme.rb]: This change has made our builds fail. Test targets are now added to the archive and are unable to code-sign? Was this intended?

As build_targets have profile, archive... all turned on?

Comment (amorde) [reply lib/xcodeproj/scheme.rb]: @joshuapoq could you submit a new issue with steps to reproduce?

Comment (dostrander) [reply lib/xcodeproj/scheme.rb]: Sorry about that, will take al ook this afternoon

Comment (joshuapoq) [reply lib/xcodeproj/scheme.rb]: No problem! I've raised an issue for this - let me know if any more info is required: https://github.com/CocoaPods/Xcodeproj/issues/618

**Pull request merged**